modules/core/src/main/scala/ch/epfl/bluebrain/nexus/rdf/GraphCursor.scala

```diff
@@ -0,0 +1,71 @@
+package ch.epfl.bluebrain.nexus.rdf
+
+import ch.epfl.bluebrain.nexus.rdf.Node.{IriNode, IriOrBNode}
+
+sealed trait GraphCursor extends Product with Serializable {
+
+  /**
+    * @return the cursor current values
+    */
+  def values: Set[Node]
+
+  /**
+    * @return the cursor current values which are of type [[IriOrBNode]]
+    */
+  def ids: Set[IriOrBNode] = values.collect { case n: IriOrBNode => n }
+
+  /**
+    * @param p the triple predicate used to test matches
+    * @param o the triple objects used to test matches
+    * @return the cursor from the available ids with the provided predicate and object functions
+    */
+  def down(p: IriNode => Boolean = _ => true, o: Node => Boolean = _ => true): GraphCursor
+}
+
+object GraphCursor {
+
+  /**
+    * An cursor which does not contain values
+    */
+  final case object EmptyCursor extends GraphCursor {
+    override def values: Set[Node]                                                                    = Set.empty
```

Comment:
I think you should insert a blank line to prevent the alignment done by scalafmt.